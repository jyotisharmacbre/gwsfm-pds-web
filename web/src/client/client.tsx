import axios from 'axios';
import authentication from '@kdpw/msal-b2c-react';
import appConfig from '../helpers/config-helper';

const config = appConfig();
export const baseURL = config.REACT_APP_MIDDLETIER_URL;
export const baseAPI = axios.create({ baseURL });
export const userServiceURL = config.REACT_APP_GATEWAY_URL;
export const userServiceAPI = axios.create({ baseURL: userServiceURL });

const isTokenHandlerEnabled = (config = {}) => {
  return config.hasOwnProperty('tokenHandlerEnabled') &&
    !config['tokenHandlerEnabled']
    ? false
    : true;
};

const isErrorHandlerEnabled = (config = {}) => {
  return config.hasOwnProperty('errorHandlerEnabled') &&
    !config['errorHandlerEnabled']
    ? false
    : true;
};

const isSuccessHandlerEnabled = (config = {}) => {
  return config.hasOwnProperty('successHandlerEnabled') &&
    !config['successHandlerEnabled']
    ? false
    : true;
};

const requestHandler = request => {
  if (isTokenHandlerEnabled(request)) {
    // Add request token here
    const token = authentication.getAccessToken();
    request.headers['accessToken'] = token;
  }
  return request;
};

baseAPI.interceptors.request.use(request => requestHandler(request));
userServiceAPI.interceptors.request.use(request => requestHandler(request));

const errorHandler = error => {
  if (isErrorHandlerEnabled(error.config)) {
    // Handle errors
  }
  return Promise.reject({ ...error });
};

const successHandler = response => {
  if (isSuccessHandlerEnabled(response.config)) {
    // Handle responses
  }
  return response;
};

baseAPI.interceptors.response.use(
  response => successHandler(response),
  error => errorHandler(error)
);

userServiceAPI.interceptors.response.use(
  response => successHandler(response),
  error => errorHandler(error)
);
