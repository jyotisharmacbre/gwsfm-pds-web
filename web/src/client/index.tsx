export { baseAPI, userServiceAPI} from './client';
