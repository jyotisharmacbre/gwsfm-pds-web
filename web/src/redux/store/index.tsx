import { applyMiddleware, createStore } from 'redux';
import { persistReducer, persistStore } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import thunk from 'redux-thunk';

import allReducer from '../reducers';

const persistConfig = {
  key: 'root',
  storage
};

const persistedReducer = persistReducer(persistConfig, allReducer);

const middleware = applyMiddleware(thunk);
const store = createStore(persistedReducer, middleware);
const persistor = persistStore(store);
export { store, persistor };
