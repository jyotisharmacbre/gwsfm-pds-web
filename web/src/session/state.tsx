export interface IProjectFormState {
    readonly projectname: string,
    readonly company: string;
    readonly customer_contract: string,
    readonly pmexperience: boolean;
    readonly projectmanager: string;
    readonly projectscope: string;
    readonly locale: string;
    readonly headofproject:string;
    readonly cnnumber:string;
    readonly comments:string;
    readonly typeofengagement:string;
    readonly currency:string;
    readonly projectowner:string;
    readonly probofwinning?:number;
    readonly approximatevalue?:number;
    readonly contracttype:string;
    readonly cdmnotifiable:boolean;
    readonly projectstatus:string;
    readonly assetworkedonprimary:string;
    readonly assetworkedonsecond:string;
    readonly assetworkedonthird:string;
    readonly soldmargin?:number;
    readonly weightedtcv?:number;
    readonly rank?:number;
    validForm: boolean;
    invalidCurrency: boolean;
    invalidProjectOwner: boolean;
    invalidProbOfWinning:boolean;
    invalidApproxValue: boolean;
    invalidContractType:boolean;
    invalidCMDNotifiable:boolean;
    invalidAssetsWorkedOnPrimary:boolean;
    invalidProjectName: boolean;
    invalidCompany: boolean;
    invalidCustomerContract: boolean;
    invalidPMExperience: boolean;
    invalidProjectScope: boolean;
    invalidLocale: boolean;
    invalidProjectManager: boolean;
    invalidHeadOfProject: boolean;
    InvalidProjectOwner: boolean;
}
