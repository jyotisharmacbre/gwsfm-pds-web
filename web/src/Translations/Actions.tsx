import { ActionCreator, Dispatch } from 'redux';
import { ThunkAction } from 'redux-thunk';
import { ILocale, IGetLocalesSuccessAction } from './Type';
import { getUserPreferences } from '../services/lookup.service';

export const getLocaleActionCreator: ActionCreator<ThunkAction<
  Promise<IGetLocalesSuccessAction>,
  ILocale,
  null,
  IGetLocalesSuccessAction
>> = () => {
  return async (dispatch: Dispatch) => {
    let res = await getUserPreferences();

    const getLocalesSuccessAction: IGetLocalesSuccessAction = {
      type: 'GetLocalesSuccess',
      locale: res.data.languageName
    };

    return dispatch(getLocalesSuccessAction);
  };
};
