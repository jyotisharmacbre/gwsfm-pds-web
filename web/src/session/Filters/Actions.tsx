import { ActionCreator, Dispatch } from "redux";
import { ThunkAction } from "redux-thunk";
import { IFilterAddAction, IFilter, IGetFilterAction } from "./Type";

export const addFilterActionCreator: ActionCreator<ThunkAction<Promise<IFilterAddAction>, IFilter, null, IFilterAddAction>> = (data: IFilter) => {
    return async (dispatch: Dispatch) => {
        const getFilterAction: IGetFilterAction = {
            type: 'GetFilterAction'
        };
        dispatch(getFilterAction);

        const addFilterAddAction: IFilterAddAction = {
            type: 'FilterAddAction',
            form: data,
        };

        return dispatch(addFilterAddAction)

    }
}