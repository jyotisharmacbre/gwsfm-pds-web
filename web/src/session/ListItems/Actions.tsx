import { ActionCreator, Dispatch } from "redux";
import { ThunkAction } from "redux-thunk";
import { IGetLocalesSuccessAction, IListItem, ListItemActions, IGetLocalesBeginAction, IGetCustomerContractSuccessAction, IGetCustomerContractBeginAction } from './Type';


export const getLocaleActionCreator: ActionCreator<ThunkAction<Promise<IGetLocalesSuccessAction>, IListItem[], null, IGetLocalesSuccessAction>> = () => {
    return async (dispatch: Dispatch) => {
        const getLocalesBeginAction: IGetLocalesBeginAction = {
            type: 'GetLocalesBegin',
        };

        dispatch(getLocalesBeginAction);

        //make api call here to return promise

        const locales = [
            {
                name: '--Please choose an option---', value: ''
            }
            ,
            {
                name: 'English', value: 'English'
            },
            {
                name: 'English (US)', value: 'English (US)'
            }
        ];

        const getLocalesSuccessAction: IGetLocalesSuccessAction = {
            type: 'GetLocalesSuccess',
            data: locales
        };
        return dispatch(getLocalesSuccessAction);
    }
}

export const getCustomerContractActionCreator: ActionCreator<ThunkAction<Promise<IGetCustomerContractSuccessAction>, IListItem[], null, IGetCustomerContractSuccessAction>> = (companyName: string) => {
    return async (dispatch: Dispatch) => {
        const getCustomerContractBeginAction: IGetCustomerContractBeginAction = {
            type: 'GetCustomerContractBegin',
        };

        dispatch(getCustomerContractBeginAction);

        //make api call here to return promise

        const cust_contracts = [
            {
                name: '---Please choose an option---', value: ''
            },
            {
                name: 'RS Electrical', value: 'RS Electrical'
            },
            {
                name: 'Demostics', value: 'Demostics'
            }
        ];
        

        const getCustomerContractSuccessAction: IGetCustomerContractSuccessAction = {
            type: 'GetCustomerContractSuccess',
            data: cust_contracts
        };
        return dispatch(getCustomerContractSuccessAction);
    }
}