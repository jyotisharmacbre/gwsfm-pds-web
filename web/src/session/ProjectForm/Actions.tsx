import { ActionCreator, Dispatch } from "redux";
import { ThunkAction } from "redux-thunk";
import { IProjectFormAddAction, IProjectForm, IGetProjectFormAction } from "./Type";
import {Validator} from "class-validator";

// Validation methods
const validator = new Validator();

export const addFormActionCreator: ActionCreator<ThunkAction<Promise<IProjectFormAddAction>, IProjectForm, null, IProjectFormAddAction>> = (data: IProjectForm) => {
    return async (dispatch: Dispatch) => {
        const getProjectFormAction: IGetProjectFormAction = {
            type: 'GetProjectFormAction'
        };
        dispatch(getProjectFormAction);

        const addProjectFormAction: IProjectFormAddAction = {
            type: 'ProjectFormAddAction',
            form: data,
            isCompanyValid: !validator.isEmpty(data.company),
            isLocaleValid: !validator.isEmpty(data.locale),
            isProjectManagerValid: !validator.isEmail(data.projectmanager),
            isProjectNameValid: !validator.isEmpty(data.projectname),
            isProjectScopeValid:!validator.isEmpty(data.projectscope)
        };

        return dispatch(addProjectFormAction)

    }
}