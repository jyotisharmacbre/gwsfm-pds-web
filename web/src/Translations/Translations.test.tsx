import nock from 'nock';
import thunk from 'redux-thunk';
import * as actions from './Actions';
import localeReducer from './Reducer';
import { baseURL } from '../client/client';
import { ILocaleState } from '../session/state';
import configureMockStore from 'redux-mock-store';
import ConnectedIntlProvider, { formatMessage } from './connectedIntlProvider';
import {
  IGetLocalesSuccessAction,
  IGetLocalesBeginAction,
  IGetLocalesFailureAction
} from './Type';
import { shallow, mount } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';
import { store } from '../store';
import { IntlProvider } from 'react-intl';
import translations from './translation';
import Translate from './translate';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

const userPreferencesMockData = {
  languageName: 'en'
};

nock(baseURL)
  .get('/api/Users/getUserPreferences')
  .reply(200, userPreferencesMockData);

describe('ConnectedIntlProvider', () => {
  it('should exist', async () => {
    // Shallow render the container passing in the mock store
    const intlProvider = shallow(
      <Provider store={mockStore()}>
        <ConnectedIntlProvider></ConnectedIntlProvider>
      </Provider>
    );
    expect(intlProvider.exists()).toBe(true);
  });

  it('should return ConnectedIntlProvider', () => {
    expect(ConnectedIntlProvider.displayName).toMatchSnapshot();
  });

});

describe('locale reducer', () => {
  let initialState: ILocaleState;
  beforeEach(() => {
    initialState = {
      locale: 'en'
    };
  });

  it('should handle GET_Locale_Begin', () => {
    const getLocalesBeginAction: IGetLocalesBeginAction = {
      type: 'GetLocalesBegin'
    };
    expect(localeReducer(undefined, getLocalesBeginAction)).toMatchSnapshot();
  });

  it('should handle GET_Locale_Success', () => {
    const getLocalesSuccessAction: IGetLocalesSuccessAction = {
      type: 'GetLocalesSuccess',
      locale: 'fr'
    };
    expect(
      localeReducer(initialState, getLocalesSuccessAction)
    ).toMatchSnapshot();
  });

  it('should handle GET_Locale_Failure', () => {
    const getLocalesFailureAction: IGetLocalesFailureAction = {
      type: 'GetLocalesFailure',
      error: { success: false }
    };
    expect(
      localeReducer(initialState, getLocalesFailureAction)
    ).toMatchSnapshot();
  });
});

describe('actions', () => {
  let store;
  beforeEach(() => {
    store = mockStore({});
  });

  it('should create an action to get locale', async () => {
    await store.dispatch(actions.getLocaleActionCreator());
    expect(store.getActions()).toMatchSnapshot();
  });
});

describe('Translation', () => {
it('should return formatted message when passing key as FIELD_VALIDATION_KEY', () => {
 mount(<Provider store={store}>
      <ConnectedIntlProvider>
      </ConnectedIntlProvider>
    </Provider>);
  expect(formatMessage('FIELD_VALIDATION_KEY', {0: 552})).toMatchSnapshot();
});

it('should return formatted message when passing a key', ()=>{

  const locale = 'en';
  const messages = translations[locale].messages
  const intlProvider = new IntlProvider({locale, messages });

  expect(Translate.getLabel( intlProvider.state.intl, 'VALIDATION_IS_REQUIRED', {0: 'Name'})).toMatchSnapshot();
})
})

