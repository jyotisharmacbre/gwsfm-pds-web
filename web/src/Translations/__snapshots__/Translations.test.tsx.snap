// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedIntlProvider should return ConnectedIntlProvider 1`] = `"Connect(IntlProvider)"`;

exports[`actions should create an action to get locale 1`] = `
Array [
  Object {
    "locale": "en",
    "type": "GetLocalesSuccess",
  },
]
`;

exports[`locale reducer should handle GET_Locale_Begin 1`] = `
Object {
  "locale": "en",
}
`;

exports[`locale reducer should handle GET_Locale_Failure 1`] = `
Object {
  "error": Object {
    "success": false,
  },
  "locale": "en",
}
`;

exports[`locale reducer should handle GET_Locale_Success 1`] = `
Object {
  "locale": "fr",
}
`;
